### ISSUE TITLE
LibreOffice macro doesn't work after saving and exiting

### ISSUE THREAD

**Issue by answerquest** [google/open-location-code#194]
I followed the visualbasic instructions for LibreOffice and it worked, I was able to run a function in my cells to generate pluscode : `=OLCENCODE(A2,B2)`

But after saving and closing the file, on opening it again, I got a macros-disabling notification, and the functions no longer worked. They simply return the latitude value.

I followed the steps [shown here](https://www.youtube.com/watch?v=sK9YR0GTH_k) to set my LibreOffice security settings to ask me whether to enable macros or not. But even after that these functions are no longer working. I get a prompt on loading the file asking about macros, I choose to enable them. But to no avail. I tried rummaging about the Organise Macros settings but LibreOffice crashed.

When I create the macro all over again, THEN it works. But I want to have people use this without having to do it all over again.

Has anyone else created a .ods (Calc) file where the OLC macro works? I even tried doing the macro in a .xlsx file, in LibreOffice. Same problem - it doesn't run things differently for .xlsx in this matter.

Environment details : Lubuntu 18.04 64-bit OS, LibreOffice 6.1.0.3

If anyone wants to see the .ods file, you can [download it here](http://nikhilvj.co.in/files/20180909-OLC_list-withMacro-notworking.ods).

**Comment by zongweil** [comment]
Assigning to @drinckes, who wrote the plugin.

**Comment by drinckes** [comment]
I did a fresh install of LibreOffice, v24.2.0.3. Started a new Calc spreadsheet, and loaded the macro as per instructions on https://github.com/google/open-location-code/tree/main/visualbasic

Put in a lat and lng and entered `=OLCEncode(A2, B2)` in a cell, and it worked. Saved and closed the sheet.

Opening the file displays `#VALUE!` in the formula cell and at the top there is the notification:

> **Macros disabled** Macros in this document are disabled due to the Macro Security settings.

I set macro security to Medium, in the Tools > Options > Security > Macro Security dialogue. Close the file, open the file, it displays a popup:

> The document contains document macros.
>
> Macros may contain viruses. Disabling macros for a document is always safe. If you disable macros you may lose functionality provided by the document macros.

Clicked Enable Macros, the sheet opens and the formulas run.

Closing as this looks like normal confusion with VB macros or maybe a bug with an old version of LibreOffice.
